provider/kubernetes/kubernetes_test.go

```diff
@@ -2163,3 +2222,28 @@ func TestProviderUpdateIngressStatus(t *testing.T) {
 	}
 
 }
+
+func TestProviderNewK8sInClusterClient(t *testing.T) {
+	p := Provider{}
+	os.Setenv("KUBERNETES_SERVICE_HOST", "localhost")
+	os.Setenv("KUBERNETES_SERVICE_PORT", "443")
+	_, err := p.newK8sClient("")
+	os.Clearenv()
```

Comment:
How about `defer`ing the cleanup right after we have set up the custom environment (i.e., right after line 2229)?

Similar below.